packages/devtools_app/lib/src/shared/ui/search.dart

```diff
@@ -893,8 +893,10 @@ class SearchField<T extends SearchControllerMixin> extends StatefulWidget {
     this.onClose,
     this.searchFieldWidth = defaultSearchFieldWidth,
     double? searchFieldHeight,
+    bool shouldExpandHeight = true,
     super.key,
-  }) : searchFieldHeight = searchFieldHeight ?? defaultTextFieldHeight;
+  }) : searchFieldHeight = searchFieldHeight ?? defaultTextFieldHeight,
+       _shouldExpandHeight = shouldExpandHeight;
```

Comment:
is there a reason this is a private field? Can shouldExpandHeight just be a public field so that we can use `this.shouldExpandHeight = true` in the constructor parameter?

Reply:
Also, should the default value be `false` to match the code below?

Reply:
Only in order to prefer private fields. https://dart.dev/effective-dart/design#prefer-making-declarations-private

It improves static analysis (like you are told when the field could be final, or is unused). It sends readability signals that the field is not intended to be read or written from outside the library. And the price you pay is that you can't use `this.` in the constructor parameter.

If there is a style in DevTools that is counter to this Effective Dart guideline, we can document that and prefer public.